openff/interchange/interop/gromacs/export/_export.py

```diff
@@ -156,30 +169,46 @@ def _get_new_entry_name(atom_type_list) -> str:
 
     def _write_moleculetypes(
         self,
-        top,
+        top: TextIOWrapper,
+        monolithic: bool,
         mapping_to_reduced_atom_types,
         merge_atom_types: bool,
     ):
         for molecule_name, molecule_type in self.system.molecule_types.items():
-            top.write("[ moleculetype ]\n")
+            # this string needs to be something that plays nicely in file paths
+            # and also works as GROMACS's label for the moleculetype "name"
+            canonicalized_name = f"{molecule_name.replace(' ', '_')}"
```

Comment:
This just replaces each occurrence of any character in the raw string `r' \/:<>"|?*'` with an underscore. Those are the bad filename characters according to a random stack overflow post I googled (plus space).

```suggestion
            canonicalized_name = molecule_name.translate({ord(c): '_' for c in r' \/:<>"|?*'})
```

I don't think the f-string is doing anything (assuming `molecule_name` is already a string)